### PR TITLE
bugfix: sliding sync initial room timelines shouldn't notify

### PR DESCRIPTION
**Pull request by kegsay**
Flag timeline events as `fromCache` when `initial: true` rooms are received. This stops notifications appearing inappropriately when you scroll the room list or spider the room list, as it causes `liveEvent=false`.

## Checklist

* [x] Tests written for new code (and old code if feasible)
* [x] Linter and other CI checks pass
* [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-js-sdk/blob/develop/CONTRIBUTING.md))

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * bugfix: sliding sync initial room timelines shouldn't notify ([\#2933](https://github.com/matrix-org/matrix-js-sdk/pull/2933)).

### CODE REVIEW

File: src/sliding-sync-sdk.ts [hunk 1]
```diff
@@ -779,7 +779,11 @@ export class SlidingSyncSdk {
         // This also needs to be done before running push rules on the events as they need
         // to be decorated with sender etc.
         room.addLiveEvents(timelineEventList, {
-            fromCache: fromCache,
+            // initial rooms are rooms that have been sent down to the client for the first time,
+            // which may include timeline events. We don't want these events to notify the user,
```
Comment (t3chguy) [inline src/sliding-sync-sdk.ts:783]: Is this true? What if a room which isn't in your sliding window pings you?

Comment (kegsay) [reply src/sliding-sync-sdk.ts:783]: baaah stop asking awkward questions :D it won't work correctly. I guess what I really want here is `initial: true` rooms _and_ the JS SDK is unaware of the room. Will play.

File: src/sliding-sync-sdk.ts [hunk 2]
```diff
@@ -725,17 +725,18 @@ export class SlidingSyncSdk {
      * @param {MatrixEvent[]} stateEventList A list of state events. This is the state
      * at the *START* of the timeline list if it is supplied.
      * @param {MatrixEvent[]} [timelineEventList] A list of timeline events. Lower index
-     * @param {boolean} fromCache whether the sync response came from cache
      * is earlier in time. Higher index is later.
+     * @param {boolean} isInitial whether timeline list is from an initial: true room.
```
Comment (t3chguy) [inline src/sliding-sync-sdk.ts:729]: `numLive` needs jsdoc

**Review by t3chguy — APPROVED** [review]

**Comment by t3chguy** [comment]
@kegsay not sure what is up with GH this time, can't find a file which defo exists, you had 100% covg before and only touched comments since, lmk if you want this force merged

**Pull request merged**